packages/airplane/internal/runtime/dev.ts

```diff
@@ -55,7 +60,40 @@ export const runtime: RuntimeInterface = {
   },
 
   sleep: async (duration: number): Promise<void> => {
-    return standardRuntime.sleep(duration);
+    const client = new Client();
+    const now = new Date();
+    const sleepID = await client.createSleep({
+      durationMs: duration,
+      until: calculateEndTimeISOString(now.getTime(), duration),
+    });
+    let isSleepFinished = false;
+    const controller = new AbortController();
+
+    const localSleep = new Promise<void>((resolve) => {
+      return sleepWithCancel(duration, controller).then(() => {
+        isSleepFinished = true;
+        resolve();
+      });
+    });
+
+    const checkSkip = async (id: string) => {
+      const poller = new Poller({ delayMs: 500 });
+      return poller.run(async () => {
+        if (isSleepFinished) {
+          return true;
+        }
+        const sleep = await client.getSleep(id);
+        if (sleep.skippedAt !== null) {
+          // Abort the timeout if the sleep was skipped.
+          controller.abort();
+          return true;
+        }
+        return null;
+      });
+    };
+
+    // Wait for either the sleep to finish or for the sleep to be skipped.
+    await Promise.all([localSleep, checkSkip(sleepID)]);
```

Comment:
Without wrapping in `Promise` above, you'll now need to call the function.

```suggestion
    await Promise.all([localSleep(), checkSkip(sleepID)]);
```